multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/DiskQuotaApplicationAttributeUpdater.java

```diff
@@ -1,25 +1,30 @@
 package org.cloudfoundry.multiapps.controller.process.util;
 
+import org.cloudfoundry.multiapps.controller.client.lib.domain.CloudApplicationExtended;
 import org.cloudfoundry.multiapps.controller.process.util.ElementUpdater.UpdateStrategy;
 
 import com.sap.cloudfoundry.client.facade.domain.CloudApplication;
+import com.sap.cloudfoundry.client.facade.domain.CloudProcess;
 
 public class DiskQuotaApplicationAttributeUpdater extends ApplicationAttributeUpdater {
 
-    public DiskQuotaApplicationAttributeUpdater(Context context) {
+    private final CloudProcess existingProcess;
+
+    public DiskQuotaApplicationAttributeUpdater(Context context, CloudProcess process) {
         super(context, UpdateStrategy.REPLACE);
+        this.existingProcess = process;
     }
 
     @Override
-    protected boolean shouldUpdateAttribute(CloudApplication existingApplication, CloudApplication application) {
+    protected boolean shouldUpdateAttribute(CloudApplication existingApplication, CloudApplicationExtended application) {
         if (application.getDiskQuota() == 0) { // 0 means "not specified".
             return false;
         }
-        return existingApplication.getDiskQuota() != application.getDiskQuota();
+        return existingProcess.getDiskInMb() != application.getDiskQuota();
```

Comment:
Does `application.getDiskQuota()` returns always number in megabyte unit?

Reply:
Good question 😄 
The `CloudApplicationExtended::getDiskQuota` is the disk quota from the MTA module.
Will check the building of the object and convert to MB if not already.

Reply:
Yes, the module disk quota is in megabytes.